src/main/java/com/happypill/application/service/order/OrderService.java

```diff
@@ -0,0 +1,133 @@
+package com.happypill.application.service.order;
+
+import com.happypill.application.auth.UserContext;
+import com.happypill.application.entity.*;
+import com.happypill.application.entity.enums.PaymentMethod;
+import com.happypill.application.exception.custom.ExceptionCode;
+import com.happypill.application.exception.global.BusinessException;
+import com.happypill.application.repository.happypilluser.HappypillUserRepository;
+import com.happypill.application.repository.order.OrderRepository;
+import com.happypill.application.repository.product.ProductRepository;
+import com.happypill.application.service.order.payment.PortonePaymentClient;
+import com.happypill.application.service.order.request.OrderCreateRequest;
+import com.happypill.application.service.order.request.OrderPaymentCompleteRequest;
+import com.happypill.application.service.order.response.OrderResponse;
+import com.happypill.application.util.SnowflakeUtil;
+import io.portone.sdk.server.payment.PaidPayment;
+import io.portone.sdk.server.payment.Payment;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class OrderService {
+
+    private final OrderRepository orderRepository;
+
+    private final ProductRepository productRepository;
+
+    private final HappypillUserRepository userRepository;
+
+    private final PortonePaymentClient portonePaymentClient;
+
+    @Transactional
+    public OrderResponse createOrder(UserContext userContext, OrderCreateRequest request) {
+        Map<Long, Product> ProductMap = getProductMapWithLock(request);
```

Comment:
```suggestion
        Map<Long, Product> productMap = getProductMapWithLock(request);
```
카멜 케이스로 수정해야할 것 같습니다

Reply:
수정완료